_posts/2022-11-11-functional-crystal.md

```diff
@@ -0,0 +1,431 @@
+---
+title: Functional Crystal
+summary: A tour on the functional aspects of Crystal
+author: beta-ziliani
+---
+
+If someone posts a claim "_Crystal is a functional language_", they would be given a grim look. I mean, classes and inheritance are all over the place in the stdlib. Surely it's an object oriented language, right?
+
+If we observe modern programming languages, we'll find that neither functional languages are strictly functional, nor object oriented ones are strictly object oriented. Or, to put it differently, these two concepts are not in contradiction; they can both be part of the tools provided by languages to abstract code. And Crystal is not an exception!
```

Comment:
"observe" can be simplified to "look at".

"Or, to put it differently" is an unneded sentence here.